File: apps/sdkwork-terminal-h5/src/bootstrap/session.ts
export interface TerminalSessionSnapshot {
  accessToken?: string;
  authToken?: string;
  refreshToken?: string;
  sessionId?: string;
}

const STORAGE_KEY = 'sdkwork-terminal.iam.session';

function readRaw(): TerminalSessionSnapshot | null {
  if (typeof window === 'undefined') {
    return null;
  }

  const legacyRaw = window.sessionStorage.getItem(STORAGE_KEY);
  const raw = window.localStorage.getItem(STORAGE_KEY) ?? legacyRaw;
  if (legacyRaw && !window.localStorage.getItem(STORAGE_KEY)) {
    window.localStorage.setItem(STORAGE_KEY, legacyRaw);
    window.sessionStorage.removeItem(STORAGE_KEY);
  }
  if (!raw) {
    return null;
  }

  try {
    return JSON.parse(raw) as TerminalSessionSnapshot;
  } catch {
    return null;
  }
}

export function readTerminalSession(): TerminalSessionSnapshot | null {
  const snapshot = readRaw();
  if (!snapshot?.authToken && !snapshot?.accessToken) {
    return null;
  }
  return snapshot;
}

export function commitTerminalSession(snapshot: TerminalSessionSnapshot): void {
  if (typeof window === 'undefined') {
    return;
  }

  window.localStorage.setItem(STORAGE_KEY, JSON.stringify(snapshot));
  window.sessionStorage.removeItem(STORAGE_KEY);
}

export function clearTerminalSession(): void {
  if (typeof window === 'undefined') {
    return;
  }

  window.sessionStorage.removeItem(STORAGE_KEY);
  window.localStorage.removeItem(STORAGE_KEY);
}

File: apps/sdkwork-terminal-flutter-mobile/lib/bootstrap/session.dart
import 'dart:convert';

import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class TerminalSessionSnapshot {
  final String? accessToken;
  final String? authToken;
  final String? refreshToken;

  const TerminalSessionSnapshot({
    this.accessToken,
    this.authToken,
    this.refreshToken,
  });

  bool get hasIamSession =>
      (authToken?.isNotEmpty ?? false) || (accessToken?.isNotEmpty ?? false);

  Map<String, String> toJson() {
    return {
      if (accessToken != null) 'accessToken': accessToken!,
      if (authToken != null) 'authToken': authToken!,
      if (refreshToken != null) 'refreshToken': refreshToken!,
    };
  }

  factory TerminalSessionSnapshot.fromJson(Map<String, dynamic> json) {
    return TerminalSessionSnapshot(
      accessToken: json['accessToken'] as String?,
      authToken: json['authToken'] as String?,
      refreshToken: json['refreshToken'] as String?,
    );
  }
}

class TerminalSessionStore {
  static const _storageKey = 'sdkwork-terminal.iam.session';
  static const _secureStorage = FlutterSecureStorage();

  static TerminalSessionSnapshot? readFromDartDefine() {
    const accessToken = String.fromEnvironment('SDKWORK_ACCESS_TOKEN');

    if (accessToken.isEmpty) {
      return null;
    }

    return TerminalSessionSnapshot(
      accessToken: accessToken,
    );
  }

  static Future<TerminalSessionSnapshot?> load() async {
    final fromDefine = readFromDartDefine();
    if (fromDefine != null) {
      return fromDefine;
    }

    final raw = await _secureStorage.read(key: _storageKey);
    if (raw == null || raw.trim().isEmpty) {
      return null;
    }

    try {
      final decoded = jsonDecode(raw) as Map<String, dynamic>;
      final snapshot = TerminalSessionSnapshot.fromJson(decoded);
      return snapshot.hasIamSession ? snapshot : null;
    } catch (_) {
      return null;
    }
  }

  static Future<void> save(TerminalSessionSnapshot snapshot) async {
    await _secureStorage.write(
      key: _storageKey,
      value: jsonEncode(snapshot.toJson()),
    );
  }

  static Future<void> clear() async {
    await _secureStorage.delete(key: _storageKey);
  }
}
